File: Easy/Temperatures/temperatures.js
const n = parseInt(readline()); // the number of temperatures to analyse
var inputs = readline().split(' ');
let closest = 6000;

if (n == 0) {
  closest = 0;
}

for (let i = 0; i < n; i++) {
    const t = parseInt(inputs[i]);// a temperature expressed as an integer ranging from -273 to 5526
    if (Math.abs(t) < Math.abs(closest)) {
        closest = t;
    } else if (t + closest == 0) {
        closest = Math.abs(t);
    }
}

console.log(closest);

File: Easy/Temperatures/temperatures.java
import java.util.*;
import java.io.*;
import java.math.*;

class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt(); 
        if(n == 0){
            System.out.println(0);
        } else {
            int result = Integer.MAX_VALUE;

            for (int i = 0; i < n; i++) {

                int t = in.nextInt();
                
                if(Math.abs(t) < Math.abs(result)){
                    result = t;
                } 

                else if(t + result == 0){
                    result = Math.abs(t);
                }         
            }

            System.out.println(result);
        }
    }
}